atlasdb-impl-shared/src/test/java/com/palantir/atlasdb/transaction/knowledge/TimestampRangeSetTest.java

```diff
@@ -90,4 +93,11 @@ public void copyAndAddAddsSeparateRangeIfDisjoint() {
                         .add(outsideRange)
                         .build());
     }
+
+    @Test
+    public void copyAndAddCopiesMinimumTimestamp() {
+        Range<Long> randomRange = Range.openClosed(50L, 55L);
+        assertThat(BASE_RANGE_SET.copyAndAdd(randomRange).minimumTimestamp())
```

Comment:
can we add following tests as well please - 
1. updating only range works (that is this one I think)
2. updating only min ts works
3. updating both works
4.

Reply:
yeah let's discuss more, I was thinking `copyAndAdd` in the end just won't take `minimumTimestamp` as a parameter (there's no point really).